test/cases/regression-composer-works-behind-satellite-fallback.sh

```diff
@@ -0,0 +1,367 @@
+   # But turn on SSL
+   SSLEngine on
+   SSLProxyEngine on
+   SSLCertificateFile ${PKI_DIR}/ca2/server.crt
+   SSLCertificateKeyFile ${PKI_DIR}/ca2/server.key
+   # And require the client to authenticate using a certificate issued by our custom CA
+   SSLVerifyClient require
+   SSLVerifyDepth 1
+   SSLCACertificateFile ${PKI_DIR}/ca2/ca.crt
+</VirtualHost>
+STOPHERE
+
+greenprint "Rewrite osbuild-composer repository configuration"
+# In case this test case runs as part of multiple different test, try not to ruit the environment
+sudo mv "${REPOSITORY_OVERRIDE}" "${REPOSITORY_OVERRIDE}.backup"
+sudo tee "${REPOSITORY_OVERRIDE}" << STOPHERE
+{
+  "${ARCH}": [
+    {
+      "baseurl": "https://localhost:8008/repo",
+      "gpgkey": "${REPO1_GPGKEY}",
+      "check_gpg": false,
+      "rhsm": true
+    },
+    {
+      "baseurl": "https://localhost:8009/repo",
+      "gpgkey": "${REPO2_GPGKEY}",
+      "check_gpg": false,
+      "rhsm": true
+    }
+  ]
+}
+STOPHERE
+
+BLUEPRINT_FILE=/tmp/bp.toml
+BLUEPRINT_NAME=zishy
+
+cat > "$BLUEPRINT_FILE" << STOPHERE
+name = "${BLUEPRINT_NAME}"
+description = "A base system with zsh"
+version = "0.0.1"
+
+[[packages]]
+name = "zsh"
+STOPHERE
+
+function try_image_build {
+    COMPOSE_START=/tmp/compose-start.json
+    COMPOSE_INFO=/tmp/compose-info.json
+    sudo composer-cli blueprints push "$BLUEPRINT_FILE"
+    if ! sudo composer-cli blueprints depsolve ${BLUEPRINT_NAME};
+    then
+        sudo cat /var/log/httpd/error_log
+        sudo journalctl -xe --unit osbuild-composer
+        exit 1
+    fi
+    if ! sudo composer-cli --json compose start ${BLUEPRINT_NAME} qcow2 | tee "${COMPOSE_START}";
+    then
+        sudo journalctl -xe --unit osbuild-composer
+        WORKER_UNIT=$(sudo systemctl list-units | grep -o -E "osbuild.*worker.*\.service")
+        sudo journalctl -xe --unit "${WORKER_UNIT}"
+        exit 1
+    fi
+    COMPOSE_ID=$(get_build_info ".build_id" "$COMPOSE_START")
+
+    # Wait for the compose to finish.
+    greenprint "⏱ Waiting for compose to finish: ${COMPOSE_ID}"
+    while true; do
+        sudo composer-cli --json compose info "${COMPOSE_ID}" | tee "${COMPOSE_INFO}" > /dev/null
+        COMPOSE_STATUS=$(get_build_info ".queue_status" "$COMPOSE_INFO")
+
+        # Is the compose finished?
+        if [[ $COMPOSE_STATUS != RUNNING ]] && [[ $COMPOSE_STATUS != WAITING ]]; then
+            break
+        fi
+
+        # Wait 30 seconds and try again.
+        sleep 30
+
+    done
+
+    sudo composer-cli compose delete "${COMPOSE_ID}" >/dev/null
+
+    sudo journalctl -xe --unit osbuild-composer
+    WORKER_UNIT=$(sudo systemctl list-units | grep -o -E "osbuild.*worker.*\.service")
+    sudo journalctl -xe --unit "${WORKER_UNIT}"
+
+    # Did the compose finish with success?
+    if [[ $COMPOSE_STATUS != FINISHED ]]; then
+        echo "Something went wrong with the compose. 😢"
+        exit 1
+    fi
+}
+
+# Part two, try that the fallback mechanism work
```

Comment:
It's not actually part two anymore so I'd change the comment a bit. Also I think it's worth keeping the check for `redhat.repo` file here and remove it if it's present. The system where this test would be ran can have `redhat.repo` present by default.

Reply:
thanks so much!